chia/full_node/full_node_api.py

```diff
@@ -1314,13 +1316,40 @@ async def request_puzzle_solution(self, request: wallet_protocol.RequestPuzzleSo
         return response_msg
 
     @api_request
-    async def request_header_blocks(self, request: wallet_protocol.RequestHeaderBlocks) -> Optional[Message]:
+    async def request_block_headers(self, request: wallet_protocol.RequestBlockHeaders) -> Optional[Message]:
         if request.end_height < request.start_height or request.end_height - request.start_height > 32:
             return None
+        height_to_hash = self.full_node.blockchain.height_to_hash
+        header_hashes: List[bytes32] = []
+        for i in range(request.start_height, request.end_height + 1):
+            header_hash: Optional[bytes32] = height_to_hash(uint32(i))
+            if header_hash is None:
+                reject = RejectBlockHeaders(request.start_height, request.end_height)
+                msg = make_msg(ProtocolMessageTypes.reject_block_headers, reject)
+                return msg
+            header_hashes.append(header_hash)
+
+        blocks_bytes: List[bytes] = await self.full_node.block_store.get_block_bytes_by_hash(header_hashes)
+        header_blocks_bytes: List[bytes] = [
+            header_block_from_block(memoryview(b), request.return_filter) for b in blocks_bytes
```

Comment:
is there a use case for the `return_filter` parameter? If we can't think of one, we probably shouldn't include it, and simplify the parser